package_bag/routines.py

```diff
@@ -0,0 +1,211 @@
+        # QUESTION: do we want to validate the json we get back?
+        # return saved json
+        return rights_json
+
+    def save_rights(self, bag, rights_json):
+        """Save JSON from rights statement service to database"""
+        bag.rights_data = rights_json
+        bag.save()
+        # save json...to file? rights.json?
+        pass
+
+
+class CreatePackage(object):
+    """Create JSON according to Ursa Major schema and package with bag"""
+
+    def run(self):
+        temp_dir = settings.TMP_DIR
+        dest_dir = settings.DEST_DIR
+        packaged = []
+        unpackaged = Bag.objects.filter(rights_data__isnull=False)
+        for u in unpackaged:
+            try:
+                self.create_json(u, temp_dir)
+                self.package_bag(temp_dir, dest_dir, u)
+                packaged.append(u)
+            except Exception as e:
+                print(e)
+        return packaged
+
+    def create_json(self, bag, temp_dir):
+        """Create JSON file to send to Ursa Major"""
+        bag_json = BagSerializer(bag).data
+        # print(bag_json)
+        with open(join(temp_dir, bag.bag_identifier, "{}.json".format(bag.bag_identifier),), "w",) as f:
+            json.dump(bag_json, f, indent=4, sort_keys=True, default=str)
+        return join(temp_dir, bag.bag_identifier,
+                    "{}.json".format(bag.bag_identifier))
+
+    def package_bag(self, temp_dir, dest_dir, bag):
+        """Create package to send to Ursa Major"""
+        tar_filename = "{}.tar.gz".format(bag.bag_identifier)
+        with tarfile.open(join(temp_dir, tar_filename), "w:gz") as tar:
+            tar.add(join(temp_dir, bag.bag_identifier),
+                    arcname=basename(join(temp_dir, bag.bag_identifier)))
+        mkdir(
+            join(dest_dir, bag.bag_identifier)
+        )
+        move(
+            join(temp_dir, tar_filename),
+            join(dest_dir, bag.bag_identifier, tar_filename,),
+        )
+
+
+class DeliverPackage(object):
+    """Deliver package to Ursa Major"""
+
+    def run(self):
+        dest_dir = settings.DEST_DIR
+        delivered = []
+        not_delivered = Bag.objects.filter(rights_data__isnull=False)
+        for d in not_delivered:
+            try:
+                self.deliver_data(d, dest_dir, settings.DELIVERY_URL)
+                delivered.append(d)
+            except Exception as e:
+                print(e)
+        return delivered
+
+    def deliver_data(self, bag, dest_dir, url):
+        """Send data to Ursa Major"""
+        bag_data = join(dest_dir, bag.bag_identifier,
+                    "{}.json".format(bag.bag_identifier))
+        r = post(
+            url,
+            json={
+                "bag_data": bag_data,
+                "origin": bag.origin,
+                "identifier": bag.bag_identifier},
+            headers={
+                "Content-Type": "application/json"},
+        )
+        r.raise_for_status()
```

Comment:
The issue with `raise_for_status()` is that it won't give you a very useful error message on its own, it will basically just give you the HTTP Error Code. You need to catch `requests.exceptions.HTTPError` so you can get a more useful error message back. See what I'm doing [here](https://github.com/RockefellerArchiveCenter/pisces/blob/master/fetcher/helpers.py#L96-L107)